src/utils/isWebExtensionManifest.js

```diff
@@ -0,0 +1,32 @@
+const path = require('path');
+const fs = require('fs');
+const logger = require('../Logger');
+
+/**
+ * Tests if a given `filename` is a WebExtension `manifest.json` file.
+ * @param {string} filename Filename to test.
+ * @returns {boolean}
+ */
+function isWebExtensionManifest(filename) {
+  if (path.basename(filename) !== 'manifest.json') {
+    return false;
+  }
+
+  try {
+    const manifestFile = fs.readFileSync(filename, {encoding: 'utf8'});
```

Comment:
Kept this `readFileSync` because using the async version would have introduced a lot of changes (`addURLDependency` becomes `async`). This code path only runs if the filename is `manifest.json`.

Reply:
What about always resulting in a shared asset in addUrlDependency, and checking in the shared asset which type of manifest this is? Than it'll definitely be async